packages/uploadthing/src/client.ts

```diff
@@ -184,11 +195,24 @@ export const DANGEROUS__uploadFiles = async <
       });
     }
 
-    const serverData = await fetch(opts.url, {
-      headers: { "x-uploadthing-polling-key": key },
-    }).then(
-      (res) => res.json() as Promise<inferEndpointOutput<TRouter[TEndpoint]>>,
-    );
+    // wait a bit as it's unsreasonable to expect the server to be done by now
+    await new Promise((r) => setTimeout(r, 750));
+
+    const serverData = (await withExponentialBackoff(async () => {
+      type PollingResponse =
+        | {
+            status: "done";
+            callbackData: inferEndpointOutput<TRouter[TEndpoint]>;
+          }
+        | { status: "still waiting" };
+
+      const res = await fetch(pollingUrl, {
+        headers: { authorization: pollingJwt },
+      }).then((r) => r.json() as Promise<PollingResponse>);
+
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-return
+      return res.status === "done" ? res.callbackData : undefined;
```

Comment:
Can't you always return `res.callbackData`? if `status !== done` it would be undefined anyway?

Reply:
ts narrowing callbackData doesnt exist on res if status isnt done, but could adjust type